### ISSUE TITLE
Is it possible to use it already with azure functions?

### ISSUE THREAD

**Issue by karladler** [flash-oss/graphql-upload-minimal#3]
Is it possible to use it already with azure functions, using the lambda or gcloud functions environment?

**Comment by koresar** [comment]
Yes.

 See the README for code examples.

**Comment by koresar** [comment]
However, I didn't yet try it in Azure.

**Comment by koresar** [comment]
I've read the question one more time. Looks like I misinterpreted it.

The right answer is - No, it does not yet support Azure.
But adding the Azure function support looks easy.

**Comment by koresar** [comment]
I've just added Azure functions support. Please, see README.
Install this: `npm i graphql-upload-minimal@next`

Code:
```js
const { processRequest } = require("graphql-upload-minimal");

exports.uploadFile = function (context, req) {
  return processRequest(context, req, { environment: "azure" });
};
```

If it works, I'll publish it as v1.1 stable.

**Comment by karladler** [comment]
awesome, I'll gonna test it the next days!

**Comment by karladler** [comment]
unfortunately I didn't managed to get it running.
Maybe I implemented it wrong, I also try to use it together with apollo server. If I upload e.g. an image it can't be opened again. 
Thats how I included it. 

`
import { buildSchema } from 'type-graphql';
import { ApolloServer } from 'apollo-server-azure-functions';
import { Context, HttpRequest } from '@azure/functions';
import { processRequest } from 'graphql-upload-minimal';

import { resolvers } from '../src/graphql';

const schema = buildSchema({
  resolvers,
  authChecker,
});

const server = new ApolloServer({
  // eslint-disable-next-line
  // @ts-ignore // eslint-disable-line
  schema,
  context: ({ context }) => context, // directly use azure Context as Apollo context,
});

const handler = server.createHandler();

// can not use async syntax here, due to problem with apollo handler
export default (context: Context, request: HttpRequest): void => {
  getConnection().then(async () => {
    try {
      const contentType = request && request.headers && request.headers['content-type'];
      if (
        typeof contentType === 'string' && contentType.includes('multipart/form-data;')
      ) {
        logger.debug('In multipart context');
        const body = await processRequest(context, request, { environment: 'azure' });
        logger.debug('###body', request.body, '####body ', body);
        request.body = body; // eslint-disable-line
        // context.req.body = body;
      }
    } catch (e) {
      logger.debug('ERRROR');
      logger.debug(e);
    }
    return handler(context, request);
  });
};
`

I already managed to have normal GraphQL requests in payload and there is a file stream generated, which I store locally.
Thats how the content of a very small png looks like:
`
�PNG


IHDR���IDAT8OՔ_kA�ϝ����F���A|�O"�K�t"�PQP��(����-IIvwf���Y�4��␅���oΙs�s���?Т(.���l�J����0*������B2%�H���k����`��"���ϳC���JTD%_<�c,��Z�lL�	kh�/*k(�9�np
l�n]�q,�W��MC�)|��6v_�<zc���A�x#�J�*V��	����}�a��FI@���БW�b�.T��0v}�Ζ§C����&Hb�V	��
{�긠��'��Cg��k�n��\�qPK� ��^���^���	��
�
��g�B�\����5��AGQ�AiJ�����
F��l��w��9aw'��R�$����)��4N��s���wn�ƕ80����+�
i��4
�"��Y!������@+�����@;Mr1�RU�XVm�'�E�ۿjٞ����u�%P�d��()�1��+����%ϵ�J�/���
tQ�5���:�I�,宛�ۿd,9��YIEND�B`�
`

**Comment by koresar** [comment]
@karladler well. The code looks legit.

But could you please also share what's the error you're seeing?

**Comment by karladler** [comment]
I don't get any error, just the png fil egets mangled and can't be openend. Some weird azure functions behavior. We solved it now with an custom parser. We will probably open source it soon.

**Comment by derSchtefan** [comment]
@koresar We run into the same issue: When we post a binary file sufficiently large, the result is that the file stream suddenly becomes about twice as big, the resulting binary file is corrupt. Is it a utf7/utf8 issue?

**Comment by koresar** [comment]
This was just fixed (as v1.2.1) thank you @derSchtefan sole efforts. Thank you!
